src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/AnalyzePackageStatus.cs

```diff
@@ -1,49 +1,44 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.IO;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.DotNet.UpgradeAssistant.Analysis;
 using Microsoft.DotNet.UpgradeAssistant.Dependencies;
 using Microsoft.Extensions.Logging;
 
 namespace Microsoft.DotNet.UpgradeAssistant.Steps.Packages
 {
     public class AnalyzePackageStatus : IAnalyzeResultProvider
     {
-        private const string SarifLogFilePath = "Dependencies.sarif";
         private readonly IDependencyAnalyzerRunner _packageAnalyzer;
         private readonly ITargetFrameworkSelector _tfmSelector;
-        private readonly IProcessResult _processResult;
         private IDependencyAnalysisState? _analysisState;
-        private IJsonSerializer _jsonSerializer;
+        private IAnalyzeResultProcessor _analyzeResultProcessor;
 
         private ILogger Logger { get; }
 
+        public string AnalysisTypeName => "Dependency Analysis";
+
         public AnalyzePackageStatus(IDependencyAnalyzerRunner packageAnalyzer,
             ITargetFrameworkSelector tfmSelector,
             ILogger<AnalyzePackageStatus> logger,
-            IProcessResult processResult,
-            IJsonSerializer jsonSerializer)
+            IAnalyzeResultProcessor analyzeResultProcessor)
```

Comment:
If you're using DI for this, then all the processors will be the same (or you'll get a list). How will you disambiguate them?

It may be easier to just new up the specific processor you need. Although, I'm not convinced it is even needed. Can this class just do the work to return a result?

Reply:
Removed the layer for IAnalyzeResultProcessor